keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -843,24 +843,53 @@ examples:
 
 ### Termination of containers
 
-In alpha sidecar containers will be terminated as regular containers. No special
+In Alpha sidecar containers will be terminated as regular containers. No special
 or additional signals will be supported.
 
-We will collect feedback from alpha implementation and plan to improve
-termination in Beta. When Pods with sidecars are terminated:
+In Beta we have thought about introducing additional termination grace period fields
+to manage termination duration
+([draft proposal](https://docs.google.com/document/d/1B01EdgWJAfkT3l6CIwNwoskQ71eyuwet3mjiQrMQbU8))
+and leverage these fields to add reverse order termination of sidecar containers
+after the primary containers terminate.
 
+However, we decided on an alternative that doesn't require additional fields or hooks while keeping
+the desired behaviors when Pods with sidecars are terminated. While original approach works better
+with truly graceful termination where consistency is more important than time taken, proposed approach
+works for that scenario as well as a more and more popular scenario of limited time to terminate when
+graceful termination is set by external requirement and Pods needs to do best to gracefully terminate
+as much as possible (think of a Spot Instances with 30 seconds notification).
+
+Here is the proposed approach:
```

Comment:
Just to fully check my understanding, for a service mesh we would:

* On preStop: signal proxy to start draining (draining == encourage clients to send elsewhere with GOAWAY/`connection: close`, etc)
* On sigterm: exit asap, we know the app container is gone

?

Reply:
Yes

Reply:
Not only are any main containers gone, but also any sidecars that started after this sidecar (e.g. put your app mesh as the first sidecar and other sidecars can rely on it and will be terminated before it).